zstacklib/zstacklib/utils/upload_task.py

```
@@ -0,0 +1,313 @@
+        up = UploadParam()
+        up.resource_uuid = headers['X-IMAGE-UUID']
+
+        up.total_size = get_long_field('X-TOTAL-SIZE')
+        up.slice_offset = get_long_field('X-SLICE-OFFSET', default=0)
+        up.slice_size = get_long_field('X-SLICE-SIZE', default=up.total_size)
+        up.slice_index = get_long_field('X-SLICE-INDEX', default=0)
+        up.slice_md5 = headers.get('X-SLICE-MD5', None)
+
+        if up.slice_offset >= up.total_size:
+            raise Exception('invalid slice offset header: %s, total_size: %d' %
+                            (up.slice_offset, up.total_size))
```

Ruff (0.13.3)

Create your own exception

(TRY002)